### PR TITLE
Allow ResultTransformer to return error at any stage

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: neo4j/driver_with_context_test.go [hunk 1]
```diff
@@ -353,6 +353,34 @@ func TestDriverExecuteQuery(outer *testing.T) {
...
+			configurers:       []ExecuteQueryConfigurationOption{ExecuteQueryWithReadersRouting()},
+			createSession: &fakeSession{
+				executeReadTransactionResult: &fakeResult{
+					nextIndex:   -1,
+					keys:        keys,
+					nextRecords: records,
+					summary:     summary,
+				}},
+			expectedSessionConfig: defaultSessionConfig,
+			expectedErr:           fmt.Errorf("accept... not"),
```
Comment (robsdedude) [inline neo4j/driver_with_context_test.go]: accept, except not :rofl:

File: neo4j/driver_with_context_test.go [hunk 2]
```diff
@@ -353,6 +353,34 @@ func TestDriverExecuteQuery(outer *testing.T) {
...
+			configurers:       []ExecuteQueryConfigurationOption{ExecuteQueryWithReadersRouting()},
+			createSession: &fakeSession{
+				executeReadTransactionResult: &fakeResult{
+					nextIndex:   -1,
+					keys:        keys,
+					nextRecords: records,
+					summary:     summary,
+				}},
+			expectedSessionConfig: defaultSessionConfig,
+			expectedErr:           fmt.Errorf("complete...ly not"),
```
Comment (robsdedude) [inline neo4j/driver_with_context_test.go:382]: :drum:

**Review by robsdedude — APPROVED** [review]
:zap: :traffic_light:

File: neo4j/driver_with_context_test.go (hunk 1 above)
Comment (fbiville) [reply neo4j/driver_with_context_test.go]: approved, let me edit the PR

**Pull request merged**